test/bot.test.js

```diff
@@ -20,24 +20,50 @@ describe('Bot', function () {
     useFakeServer: false
   });
 
+  const createGuildStub = () => ({
+    roles: new discord.Collection(),
+    members: new discord.Collection(),
+    emojis: new discord.Collection()
+  });
+
   beforeEach(function () {
     this.infoSpy = sandbox.stub(logger, 'info');
     this.debugSpy = sandbox.stub(logger, 'debug');
     this.errorSpy = sandbox.stub(logger, 'error');
     this.sendStub = sandbox.stub();
-    this.findUserStub = sandbox.stub();
-    this.findRoleStub = sandbox.stub();
-    this.findEmojiStub = sandbox.stub();
+
+    this.discordUsers = new discord.Collection();
     irc.Client = ClientStub;
-    discord.Client = createDiscordStub(
-      this.sendStub, this.findUserStub, this.findRoleStub, this.findEmojiStub
-    );
+    this.guild = createGuildStub();
+    discord.Client = createDiscordStub(this.sendStub, this.guild, this.discordUsers);
 
     ClientStub.prototype.say = sandbox.stub();
     ClientStub.prototype.send = sandbox.stub();
     ClientStub.prototype.join = sandbox.stub();
     this.bot = new Bot(config);
     this.bot.connect();
+
+    this.addUser = function (user, member = null) {
+      const userObj = new discord.User(this.bot.discord, user);
+      const guildMember = Object.assign(member || user, { user: userObj });
```

Comment:
This will mutate either `member` or `user`, might want to do `Object.assign({}, member || user, { user: userObj })` instead or use object spread (not that it matters too much, but mutating arguments leads to funky bugs from time to time)